SnapWar/ViewControllers/SWAuthViewController.swift

```diff
@@ -28,21 +29,66 @@ class SWAuthViewController: UIViewController {
         formContainer.isLayoutMarginsRelativeArrangement = true
         formContainer.layoutMargins = UIEdgeInsets(top: 0, left: horizontalMargin, bottom: 0, right: horizontalMargin)
         
-        let keyboardTypes = [
-            UIKeyboardType.alphabet,
-            UIKeyboardType.numberPad,
-            UIKeyboardType.emailAddress
-        ]
-        for keyboardType in keyboardTypes {
-            let textField = SWTextField()
-            textField.keyboardType = keyboardType
-            textField.placeholder = "type here"
+        usernameField.placeholder = "username"
+        usernameField.delegate = self
+        usernameField.tag = 0
+        formContainer.addArrangedSubview(usernameField)
 
-            formContainer.addArrangedSubview(textField)
-        }
-        
-        let passwordField = SWPasswordTextField()
         passwordField.placeholder = "password"
+        passwordField.delegate = self
```

Comment:
oh yeah, forgot about it, but it's good practice to set the delegate to nil under the lifecycle method `viewWillDisappear`. this could help prevent any potential circular memory references that could lead to memory leaks